src/main/java/io/reactiverse/pgclient/impl/codec/DataType.java

```diff
@@ -99,7 +139,40 @@ public void encodeBinaryInternal(Short value, ByteBuf buff) {
     }
   };
 
-  public static DataType<short[]> INT2_ARRAY = new DataType<>(short[].class, 1005);
+  public static final DataType<short[]> INT2_ARRAY = new DataType<short[]>(short[].class, 1005) {
+
+    @Override
+    public short[] decodeBinary(int len, ByteBuf buff) {
+      if (len == 12) {
+        return new short[0];
```

Comment:
use a single instance of zero length array since they are immutable

Reply:
That works for the primitive types but not for the Objects which use the ArrayFactory. Or am I missing something?